### PR TITLE
Final repo upgrade

### PR DESCRIPTION
**Pull request by paritshivani**

### CODE REVIEW

File: projects/event-library/src/lib/events/components/cover-event-detail/cover-event-detail.component.ts [hunk 1]
```diff
@@ -14,25 +14,34 @@ import { UserConfigService } from '../../services/userConfig/user-config.service
...
   ngOnInit() {
-      this.userConfigService.userId = this.userData;
-      this.isEnrolledToEvent();
+    // this.dataService.get({url : 'https://jsonplaceholder.typicode.com/todos/1'}).subscribe(response => 
+    //   {
+    //     console.log({response});
+    //   });
+
+    setTimeout(() => {
+     this.isOwner = (this.eventDetailItem.owner == this.userData) ? true : false;
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/cover-event-detail/cover-event-detail.component.ts:40]: @paritshivani Remove this set timeout. Move this code subscribe block of eventDetailItem

File: projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts [hunk 2]
```diff
@@ -20,18 +22,25 @@ export class JoinEventComponent implements OnInit {
...
-    public translate: TranslateService,
     private eventService: EventService,
-    private timezoneCal: TimezoneCal) {
+    private timezoneCal: TimezoneCal
+        ) {
   }
 
   ngOnInit() {
+    setTimeout(() => {
       this.isEnrollEvent();
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts:34]: @paritshivani  Remove setTimeout.

File: projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts [hunk 3]
```diff
@@ -20,18 +22,25 @@ export class JoinEventComponent implements OnInit {
...
+
+    }, 1000);
+
+    setInterval(() => {
```
Comment (ahghatol) [inline projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts:38]: @paritshivani Why are you calling join event button after reach second?

File: projects/event-library/src/test.ts [hunk 4]
```diff
@@ -16,6 +16,8 @@ getTestBed().initTestEnvironment(
   platformBrowserDynamicTesting()
 );
 // Then we find all the tests.
-const context = require.context('./', true, /\.spec\.ts$/);
+//const context = require.context('./', true, /\.spec\.ts$/);
```
Comment (ahghatol) [inline projects/event-library/src/test.ts:19]: it should run all components specs

File: server.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+app.set('port', 3000);
+app.use(bodyParser.urlencoded({ extended: false }))
+app.use(bodyParser.json({ limit: '50mb' }))
+
+app.get("/latex/convert", latexService.convert)
+app.post("/latex/convert", bodyParser.json({ limit: '1mb' }), latexService.convert);
+
+
+app.all(['/api/framework/v1/read/*', '/learner/framework/v1/read/*', '/api/channel/v1/read/*'], proxy('dev.sunbirded.org', {
+    https: true,
```
Comment (ahghatol) [inline server.js:19]: @paritshivani  Remove the unused code

File: server.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            // you can update headers
+        proxyReqOpts.headers['Content-Type'] = 'application/json';
+        proxyReqOpts.headers['user-id'] = 'content-editor';
+        proxyReqOpts.headers['authorization'] = 'Bearer ';
+        return proxyReqOpts;
+    }
+}));
+
+
+app.use(['/api', '/assets', '/action'], proxy('staging.sunbirded.org', {
```
Comment (ahghatol) [inline server.js:35]: take the host from env var file, don't harcode it

File: src/app/app-routing.module.ts [hunk 7]
```diff
@@ -6,14 +6,18 @@ import { DetailRedirectionComponent } from'./detail-redirection/detail-redirecti
 import { UserListComponent } from './user-list/user-list.component';
 import { UserFomComponent } from './user-fom/user-fom.component';
 import { UserDetailComponent } from './user-detail/user-detail.component';
+import { DemoComponent } from '../app/demo/demo.component';
```
Comment (ahghatol) [inline src/app/app-routing.module.ts:9]: What is the demo component here?

File: src/app/detail-redirection/detail-redirection.component.ts [hunk 8]
```diff
@@ -26,7 +26,7 @@ export class DetailRedirectionComponent implements OnInit {
   }
 
   showEventDetailPage() {
-    this.eventDetailService.getEvent().subscribe((data: any) => {
+    this.eventDetailService.getEvent('123').subscribe((data: any) => {
```
Comment (ahghatol) [inline src/app/detail-redirection/detail-redirection.component.ts:29]: Get the get id from query param

File: projects/event-library/src/lib/events/components/cover-event-detail/cover-event-detail.component.ts (hunk 1 above)
Comment (paritshivani) [reply projects/event-library/src/lib/events/components/cover-event-detail/cover-event-detail.component.ts:40]: eventDetailItem is taken from @Input and it takes time to receive it via selector

File: projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts (hunk 2 above)
Comment (paritshivani) [reply projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts:34]: here also details are taken from @input and it takes time to receive it via selector

File: projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts (hunk 3 above)
Comment (paritshivani) [reply projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts:38]: details are taken from @input and it takes time to receive it via selector. I have commented the timeOut  for now. Will find the solution for this.

File: src/app/app-routing.module.ts (hunk 7 above)
Comment (paritshivani) [reply src/app/app-routing.module.ts:9]: We have ceated new demo component to demonstrate all list, details, crate and calender view at once for NSDL weekly demos. Will remove later

File: projects/event-library/src/lib/events/components/cover-event-detail/cover-event-detail.component.ts (hunk 1 above)
Comment (ahghatol) [reply projects/event-library/src/lib/events/components/cover-event-detail/cover-event-detail.component.ts:40]: Angular by default takes care of it, please fix it 

Refer: https://stackoverflow.com/questions/38470540/angular2-subscribing-to-changes-to-input-in-child-component

File: projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts (hunk 2 above)
Comment (ahghatol) [reply projects/event-library/src/lib/events/components/join-event-button/join-event-button.component.ts:34]: Refer prev comment

File: server.js (hunk 5 above)
Comment (ahghatol) [reply server.js]: Is this fixed

**Pull request merged**